app/server/src/main/scala/org/bitcoins/server/BitcoinSServerMain.scala

```diff
@@ -33,177 +34,180 @@ class BitcoinSServerMain(override val args: Array[String])
   implicit lazy val conf: BitcoinSAppConfig =
     BitcoinSAppConfig(datadir, baseConfig)
 
-  def startup: Future[Unit] = {
+  implicit lazy val walletConf: WalletAppConfig = conf.walletConf
+  implicit lazy val nodeConf: NodeAppConfig = conf.nodeConf
+  implicit lazy val chainConf: ChainAppConfig = conf.chainConf
+  implicit lazy val bitcoindRpcConf: BitcoindRpcAppConfig = conf.bitcoindRpcConf
 
-    implicit val walletConf: WalletAppConfig = conf.walletConf
-    implicit val nodeConf: NodeAppConfig = conf.nodeConf
-    implicit val chainConf: ChainAppConfig = conf.chainConf
-    implicit val bitcoindRpcConf: BitcoindRpcAppConfig = conf.bitcoindRpcConf
+  override def start(): Future[Unit] = {
+    val startFut = nodeConf.nodeType match {
+      case _: InternalImplementationNodeType =>
+        startBitcoinSBackend()
+      case NodeType.BitcoindBackend =>
+        startBitcoindBackend()
+    }
 
-    def startBitcoinSBackend(): Future[Unit] = {
-      if (nodeConf.peers.isEmpty) {
-        throw new IllegalArgumentException(
-          "No peers specified, unable to start node")
-      }
+    startFut.failed.foreach { err =>
+      logger.error(s"Error on server startup!", err)
+      err.printStackTrace()
+      throw err
+    }
+    startFut
+  }
 
-      val peerSocket =
-        NetworkUtil.parseInetSocketAddress(nodeConf.peers.head,
-                                           nodeConf.network.port)
-      val peer = Peer.fromSocket(peerSocket)
+  override def stop(): Future[Unit] = {
+    logger.error(s"Exiting process")
+    for {
+      _ <- walletConf.stop()
+      _ <- nodeConf.stop()
+      _ <- chainConf.stop()
+      _ = logger.info(s"Stopped ${nodeConf.nodeType.shortName} node")
+      _ <- system.terminate()
+    } yield {
+      logger.info(s"Actor system terminated")
+      ()
+    }
+  }
 
-      //initialize the config, run migrations
-      val configInitializedF = conf.start()
+  def startBitcoinSBackend(): Future[Unit] = {
```

Comment:
The reason this diff is so big is because i had to move `startBitcoinSBackend()` and `startBitcoindBackend()` out of the inside of `startup`